docs/rfcs/builder-config-wildcard.md

```diff
@@ -0,0 +1,153 @@
+# OpenTelemetry Wildcard Name Matching
+
+## Overview
+
+### What and Why
+
+The following is a specification for OpenTelemetry Wildcard Name Matching grammar that could be added to the generated `metadata` packages from `mdatagen`. This grammar will facilitate the use of wildcards in metric and resource attribute configurations to enabling/disabling metrics/resource attributes in a more concise way, allowing a configuration like:
```

Comment:
Arguably this could be added as a package in the collector repo that the templates import.